src/main/test/mate/academy/dao/impl/UserDaoImplTest.java

```diff
@@ -0,0 +1,91 @@
+package mate.academy.dao.impl;
+
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.dao.AbstractTest;
+import mate.academy.dao.RoleDao;
+import mate.academy.dao.UserDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class UserDaoImplTest extends AbstractTest {
+    private static final String EMAIL = "alice@i.ua";
+    private static final String PASSWORD = "12345678";
+    private static final Long ID = 1L;
+    private UserDao userDao;
+    private User user;
+    private Role role;
+
+    @Override
+    protected Class<?>[] entities() {
+        return new Class[]{User.class, Role.class};
+    }
+
+    @BeforeEach
+    public void setUp() {
+        userDao = new UserDaoImpl(getSessionFactory());
+        RoleDao roleDao = new RoleDaoImpl(getSessionFactory());
+        role = roleDao.save(new Role(Role.RoleName.USER));
+        user = userDao.save(testUser());
+    }
+
+    @Test
+    public void save_Ok() {
+        assertNotNull(user);
+        assertEquals(ID, user.getId());
+        assertEquals(EMAIL, user.getEmail());
+    }
```

Comment:
you are basically performing a save test inside the `setUp()` method, it should not be like that, `before each` methods are only for creating some test, not-saved data, especially if you are testing the dao itself in this class